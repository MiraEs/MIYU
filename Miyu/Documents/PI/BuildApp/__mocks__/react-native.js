/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
