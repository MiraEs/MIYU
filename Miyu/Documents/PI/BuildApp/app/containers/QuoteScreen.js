import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	ActionSheetIOS,
	Alert,
	StyleSheet,
	TouchableOpacity,
	View,
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import styles from '../lib/styles';
import {
	Button,
	Image,
	ListView,
	withScreen,
	Text,
} from 'BuildLibrary';
import { getQuoteComposite } from '../actions/QuotesActions';
import { loadQuote } from '../actions/CartActions';
import helpers from '../lib/helpers';
import TrackingActions from '../lib/analytics/TrackingActions';
import EventEmitter from '../lib/eventEmitter';
import {
	IMAGE_75,
	IMAGE_100,
} from '../constants/CloudinaryConstants';
import ExpandableListItem from '../components/ExpandableCartItem';
import environment from '../lib/environment';
import ContactUsHelper from '../lib/ContactUsHelper';
import { showAlert } from '../actions/AlertActions';
import PhoneHelper from '../lib/PhoneHelper';

const componentStyles = StyleSheet.create({
	bottomBorder: {
		borderColor: styles.colors.greyLight,
		borderBottomWidth: styles.dimensions.borderWidth,
		paddingBottom: styles.measurements.gridSpace1,
		marginBottom: styles.measurements.gridSpace1,
	},
	headerDetails: {
		marginHorizontal: styles.measurements.gridSpace1,
	},
	bottomLayout: {
		backgroundColor: styles.colors.white,
		borderColor: styles.colors.grey,
		borderTopWidth: styles.dimensions.borderWidth,
	},
	estimatedTotal: {
		flexDirection: 'row',
		padding: styles.measurements.gridSpace2,
		justifyContent: 'space-between',
	},
	header: {
		backgroundColor: styles.colors.white,
		flexDirection: 'row',
		padding: styles.measurements.gridSpace1,
		marginBottom: styles.measurements.gridSpace1,
	},
	imageWrapper: {
		borderWidth: styles.dimensions.borderWidth,
		borderColor: styles.colors.lightGray,
		borderRadius: IMAGE_75.width / 2,
		marginTop: styles.measurements.gridSpace1,
		marginHorizontal: styles.measurements.gridSpace1,
	},
	screen: {
		flex: 1,
		backgroundColor: styles.colors.greyLight,
	},
});

export class QuoteScreen extends Component {

	constructor(props) {
		super(props);
		this.state = {
			loadingCart: false,
			dataSource: new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 }),
		};
	}


	componentWillReceiveProps({ composite }) {
		if (composite) {
			const { quote } = composite;
			if (composite !== this.props.composite && quote && !quote.isValid) {
				Alert.alert(
					'Cart not valid',
					`This cart is no longer valid because it has ${Date.now() > quote.expirationDate ? 'expired' : 'already been purchased'}.`,
					[{
						text: 'OK',
						onPress: () => this.props.navigator.pop(),
					}], {
						cancelable: false,
					},
				);
			} else if (composite.sessionCart) {
				this.setState({
					dataSource: this.state.dataSource.cloneWithRows(composite.sessionCart.sessionCartItems),
				});
			}
		}
	}

	setScreenTrackingInformation() {
		return {
			name: 'build:app:quotes',
		};
	}

	getScreenData = () => {
		this.props.actions.getQuoteComposite(this.props.quoteNumber);
	};

	renderHeaderTotal = () => {
		const { quote, sessionCart } = this.props.composite;
		if (!quote.hidePricing) {
			return (
				<Text size="small">Estimated Total: <Text
					weight="bold"
					size="small"
				>{helpers.toUSD(sessionCart.subTotal)}</Text></Text>
			);
		}
	};

	renderFooterTotal = () => {
		const { quote, sessionCart } = this.props.composite;
		if (!quote.hidePricing && sessionCart) {
			return (
				<View style={componentStyles.estimatedTotal}>
					<Text
						lineHeight={false}
						weight="bold"
					>
						Estimated Total
					</Text>
					<Text
						lineHeight={false}
						color="primary"
						size="large"
						weight="bold"
					>
						{helpers.toUSD(sessionCart.subTotal)}
					</Text>
				</View>
			);
		}
	};

	onPressLoadCart = () => {
		const cartOptions = {
			quoteNumber: this.props.quoteNumber,
		};
		const onPress = (options) => {
			if (!this.state.loadingCart) {
				this.setState({ loadingCart: true });
				this.props.actions.loadQuote(options)
				.then(() => {
					this.props.navigation.getNavigator('root').push('cartScreen', {
						title: 'Shopping Cart',
					});
					this.props.navigator.pop();
				})
				.catch(() => {
					this.setState({ loadingCart: false });
					this.props.actions.showAlert('Unable to load cart.', 'error');
				})
				.done();
			}
		};
		const buttons = [{
			text: 'Discard Current Cart',
			onPress: () => onPress(cartOptions),
		}, {
			text: 'Merge with Current Cart',
			onPress: () => onPress({
				...cartOptions,
				sessionCartId: this.props.cart.sessionCartId,
			}),
		}, {
			text: 'Cancel',
			onPress: helpers.noop,
		}];
		const { cart } = this.props;
		if (cart.sessionCartItems && !cart.sessionCartItems.length) {
			onPress(cartOptions);
		} else {
			if (helpers.isIOS()) {
				ActionSheetIOS.showActionSheetWithOptions({
					options: buttons.map((button) => button.text),
					cancelButtonIndex: buttons.length - 1,
					tintColor: styles.colors.secondary,
				}, (index) => buttons[index].onPress());
			} else {
				delete buttons[buttons.length - 1];
				EventEmitter.emit('showActionSheet', {
					title: 'Load Cart',
					options: buttons,
				});
			}
		}
	};

	renderProjectName = (quote) => {
		if (quote.projectTitle) {
			return (
				<Text
					weight="bold"
					size="small"
				>
					{quote.projectTitle}
				</Text>
			);
		}
	};

	renderPhoneNumber = (quote) => {
		if (quote.phoneNumber && quote.employeeId) {
			const phone = {
				phoneNumber: quote.phoneNumber,
			};
			return (
				<TouchableOpacity onPress={() => ContactUsHelper.callUs(phone)}>
					<Text
						color="primary"
						weight="bold"
						size="small"
					>
						{PhoneHelper.formatPhoneNumber(phone)}
					</Text>
				</TouchableOpacity>
			);
		}
	};

	renderName = (quote) => {
		if (quote.externalSenderName || quote.employeeFullName) {
			return (
				<Text
					size="small"
					weight="bold"
				>
					{quote.externalSenderName || quote.employeeFullName}
				</Text>
			);
		}
	};

	renderEmail = (quote) => {
		const email = quote.employeeEmail || quote.externalSenderEmail;
		if (email) {
			return (
				<TouchableOpacity onPress={() => helpers.emailUs(email)}>
					<Text
						size="small"
						color="primary"
						weight="bold"
					>
						{email}
					</Text>
				</TouchableOpacity>
			);
		}
	};

	renderHeaderImage = (quote) => {
		const source = {
			uri: helpers.getProfileImage({ repUserId: quote.employeeId }),
		};
		return (
			<Image
				resizeMode="contain"
				source={source}
				{...IMAGE_100}
			/>
		);
	};

	renderHeader = () => {
		const { composite: { sessionCart, quote } } = this.props;
		if (sessionCart && quote) {
			return (
				<View style={componentStyles.header}>
					{this.renderHeaderImage(quote)}
					<View>
						<View style={[componentStyles.headerDetails, componentStyles.bottomBorder]}>
							<Text size="small">Cart Number: <Text
								weight="bold"
								size="small"
							>{quote.cartNumber}</Text></Text>
							{this.renderProjectName(quote)}
							<Text size="small">Total Items: <Text
								weight="bold"
								size="small"
							>{sessionCart.quantity}</Text></Text>
							{this.renderHeaderTotal()}
						</View>
						<View style={componentStyles.headerDetails}>
							{this.renderName(quote)}
							{this.renderEmail(quote)}
							{this.renderPhoneNumber(quote)}
							<Text size="small">Created: <Text
								weight="bold"
								size="small"
							>{helpers.getDateStrictFormat(quote.quoteCreateDate)}
							</Text></Text>
							<Text size="small">Exp: <Text
								weight="bold"
								size="small"
							>{helpers.getDateStrictFormat(quote.expirationDate)}</Text></Text>
						</View>
					</View>
				</View>
			);
		}
	};

	renderSeparator = (sectionId, rowId) => {
		const { sessionCartItems } = this.props.composite.sessionCart;
		const style = {
			height: rowId < sessionCartItems.length - 1 ? styles.measurements.gridSpace1 : 0,
		};
		return (
			<View
				key={rowId}
				style={style}
			/>
		);
	};

	renderButton = () => {
		const { hidePricing, onlineAllowed } = this.props.composite.quote;
		if (hidePricing || !onlineAllowed) {
			return (
				<Button
					isLoading={this.state.loadingCart}
					text="Call to Order"
					onPress={() => ContactUsHelper.callUs({ phoneNumber: environment.phone })}
					trackAction={TrackingActions.QUOTE_CALL_TO_ORDER}
					accessibilityLabel="Call to Order Button"
				/>
			);
		}
		return (
			<Button
				isLoading={this.state.loadingCart}
				text="Load Cart"
				onPress={this.onPressLoadCart}
				trackAction={TrackingActions.QUOTE_LOAD_CART}
				accessibilityLabel="Load Cart Button"
			/>
		);
	};

	renderScreenContent = () => {
		const { composite, error } = this.props;
		if (composite) {
			return (
				<View style={styles.elements.flex1}>
					<ListView
						renderHeader={this.renderHeader}
						dataSource={this.state.dataSource}
						renderRow={(item) => (
							<ExpandableListItem
								hidePricing={composite.quote.hidePricing}
								item={item}
							/>
						)}
						renderSeparator={this.renderSeparator}
					/>
					<View style={componentStyles.bottomLayout}>
						{this.renderFooterTotal()}
						{this.renderButton()}
					</View>
				</View>
			);
		}
		if (error) {
			return <Text textAlign="center">{this.props.error}</Text>;
		}
	};

	render() {
		return (
			<View style={styles.elements.screenGreyLight}>
				<View
					style={componentStyles.screen}
				>
					{this.renderScreenContent()}
				</View>
			</View>
		);
	}

}

QuoteScreen.route = {
	navigationBar: {
		visible: true,
		title: 'Saved Cart Preview',
	},
};

const mapStateToProps = (state, ownProps) => {
	return {
		cart: state.cartReducer.cart,
		composite: state.quotesReducer.composites[ownProps.quoteNumber],
		error: state.quotesReducer.error,
		loading: !state.quotesReducer.composites.hasOwnProperty(ownProps.quoteNumber) && !state.quotesReducer.error,
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({
			loadQuote,
			getQuoteComposite,
			showAlert,
		}, dispatch),
	};
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(withScreen(QuoteScreen));

QuoteScreen.propTypes = {
	actions: PropTypes.object,
	cart: PropTypes.object,
	error: PropTypes.string,
	loading: PropTypes.bool,
	title: PropTypes.string,
	navigation: PropTypes.shape({
		getNavigator: PropTypes.func,
	}),
	navigator: PropTypes.shape({
		pop: PropTypes.func,
		replace: PropTypes.func,
	}),
	composite: PropTypes.shape({
		quote: PropTypes.object,
		sessionCart: PropTypes.object,
	}),
	quoteNumber: PropTypes.string.isRequired,
};
