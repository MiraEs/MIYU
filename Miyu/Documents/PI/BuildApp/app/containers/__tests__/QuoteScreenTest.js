jest.unmock('react-native');
import 'react-native';
import React from 'react';
import { QuoteScreen } from '../QuoteScreen';

jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('BuildLibrary');
jest.mock('BuildNative');
jest.mock('../../../app/components/NavigationBar', () => 'NavigationBar');
jest.mock('../../../app/components/FavoriteButton', () => 'FavoriteButton');
jest.mock('../../../app/components/button', () => 'Button');
jest.mock('../../../app/components/Cart/CartRow', () => 'CartRow');
jest.mock('../../../app/containers/CartScreen', () => 'CartScreen');

const defaultProps = {
	actions: {
		trackState: jest.fn(),
	},
	composite: {
		quote: {
			cartNumber: 12345,
			employeeId: 12345,
			employeeEmail: 'test@build.com',
			employeeFullName: 'Test',
			expirationDate: 1,
			hidePricing: false,
			onlineAllowed: true,
			phoneNumber: '1231231234',
			projectTitle: 'test',
			quoteCreateDate: 1,
		},
		sessionCart: {
			quantity: 1,
			sessionCartItems: [],
			subTotal: 1,
		},
	},
	loading: false,
	title: '',
	quoteNumber: '1',
};

describe('QuotesScreen container', () => {

	it('should render correctly with no quote number', () => {
		const tree = require('react-test-renderer').create(
			<QuoteScreen {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
