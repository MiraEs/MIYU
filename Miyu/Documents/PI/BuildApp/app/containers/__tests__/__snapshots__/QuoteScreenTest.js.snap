// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotesScreen container should render correctly with no quote number 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EBEDEF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#EBEDEF",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <ListView
        dataSource={
          Object {
            "cloneWithRows": [Function],
            "cloneWithRowsAndSections": [Function],
          }
        }
        renderHeader={[Function]}
        renderRow={[Function]}
        renderSeparator={[Function]}
      />
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D0D3D4",
            "borderTopWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
            }
          }
        >
          <Text
            lineHeight={false}
            weight="bold"
          >
            Estimated Total
          </Text>
          <Text
            color="primary"
            lineHeight={false}
            size="large"
            weight="bold"
          >
            $1.00
          </Text>
        </View>
        <Button
          accessibilityLabel="Load Cart Button"
          isLoading={false}
          onPress={[Function]}
          text="Load Cart"
          trackAction="build:app:quotescreen:loadcart"
        />
      </View>
    </View>
  </View>
</View>
`;
